animal_crossing.py

```diff
@@ -36,11 +38,90 @@ def stalk_market(cmd):
 		_stalk_set_timezone(cmd, subargs)
 	elif subcmd == 'sell':
 		_stalk_set_sell_price(cmd, subargs)
+	elif subcmd == 'buy':
+		_stalk_set_buy_price(cmd, subargs)
 	elif subcmd == 'trigger':
 		_stalks_set_sell_trigger(cmd, subargs)
 	else:
 		cmd.reply('Unrecognized stalks subcommand %s' % subcmd)
 
+def _stalk_set_buy_price(cmd, price):
+	if not price:
+		_stalk_list_buy_prices(cmd)
+		return
+
+	user_id = cmd.sender['id']
+	current_time = datetime.datetime.now(datetime.timezone.utc)
+	cur = db.execute('SELECT timezone FROM user WHERE id = ?', (user_id,))
+	res = cur.fetchone()
+	if not res:
+		cmd.reply('Could not add buy price. Have you registered a friend code?')
+		return
+	elif res['timezone'] is None:
+		cmd.reply('Could not add buy price. Please register a time zone with !stalks tz')
+		return
+
+	user_time = current_time.astimezone(dateutil.tz.gettz(res['timezone']))
+	if user_time.weekday() != 6:
+		cmd.reply('It is not currently Sunday in your selected time zone, %s. Turnip offers cannot be submitted.' %
+			res['timezone'])
+		return
+	elif user_time.hour >= 12:
+		cmd.reply('Turnips are not available on your island. Your current time zone is %s, where it is currently %s.' %
+			(res['timezone'], user_time.strftime(time_format)))
+		return
+	try:
+		value = int(price)
+	except ValueError:
+		cmd.reply('Could not parse buy value. Usage: !stalks buy 123')
+		return
+
+	week_local, _, expiration = _user_time_info(user_time)
+	with db:
+		db.execute('''
+		INSERT INTO buy_price (user_id, week_local, expiration, price) VALUES (?, ?, ?, ?)
+		''', (user_id, week_local, expiration.astimezone(datetime.timezone.utc), value))
+
+	expires_in = readable_rel(expiration - user_time)
+	cmd.reply('Buy price recorded at %d bells. Offer expires in %s.\n<%s>' %
+		(value, expires_in, _turnip_prophet([value] + [None] * 12)))
+
+def _stalk_list_buy_prices(cmd):
+	current_time = datetime.datetime.now(datetime.timezone.utc)
+	sunday = _date_to_sunday(current_time)
```

Comment:
lol uh, bad news. `_date_to_sunday` subtracts `datetime.datetime.isoweekday` days which returns 7 for sunday so it records `week_local` as last week